lib/browser/mqtt.ts

```diff
@@ -337,13 +335,15 @@ export class MqttClientConnection extends BufferedEventEmitter {
      */
     async publish(topic: string, payload: Payload, qos: QoS, retain: boolean = false): Promise<MqttRequest> {
         let payload_data = normalize_payload(payload);
-        return this.connection.publish(topic, payload_data, { qos: qos, retain: retain })
-            .catch((reason) => {
-                this.emit('error', new CrtError(reason));
-            })
-            .then((value) => {
-                return { packet_id: (value as IPublishPacket).messageId };
+        return new Promise((resolve, reject) => {
+            this.connection.publish(topic, payload_data, { qos: qos, retain: retain }, (error, packet) => {
+                if (error) {
+                    reject(new CrtError(error));
+                    return this.on_error(error);
```

Comment:
why `return` here? on_error doesn't return anything, and this is a callback where the caller doesn't expect a return value

Are you doing this just to early-out? If that's the case, a lone `return;` would be less confusing. or just use an else{} block

Reply:
This is what TS/JS people do 🤷‍♂️ . It's just an early out.

Reply:
And here I was, with all this respect for JS now that I've learned how cool the modern ecosystem is

and now it's all gone

j/k ok whatever